src/credentials/attestation_verifier/TestDACRevocationDelegateImpl.cpp

```diff
@@ -0,0 +1,205 @@
+
+    return BytesToHexStr(akid, outAKIDHexStr);
+}
+
+CHIP_ERROR TestDACRevocationDelegateImpl::GetSerialNumberHexStr(const ByteSpan & certDer, MutableCharSpan & outSerialNumberHexStr)
+{
+    uint8_t serialNumberBuf[kMaxCertificateSerialNumberLength] = { 0 };
+    MutableByteSpan serialNumber(serialNumberBuf);
+
+    ReturnErrorOnFailure(ExtractSerialNumberFromX509Cert(certDer, serialNumber));
+    return BytesToHexStr(serialNumber, outSerialNumberHexStr);
+}
+
+CHIP_ERROR TestDACRevocationDelegateImpl::GetIssuerNameBase64Str(const ByteSpan & certDer,
+                                                                 MutableCharSpan & outIssuerNameBase64String)
+{
+    uint8_t issuerBuf[kMaxCertificateDistinguishedNameLength] = { 0 };
+    MutableByteSpan issuer(issuerBuf);
+
+    ReturnErrorOnFailure(ExtractIssuerFromX509Cert(certDer, issuer));
+    VerifyOrReturnError(outIssuerNameBase64String.size() >= BASE64_ENCODED_LEN(issuer.size()), CHIP_ERROR_BUFFER_TOO_SMALL);
+
+    uint32_t encodedLen = Base64Encode32(issuer.data(), static_cast<uint32_t>(issuer.size()), outIssuerNameBase64String.data());
+    outIssuerNameBase64String.reduce_size(encodedLen);
+    return CHIP_NO_ERROR;
+}
+
+bool TestDACRevocationDelegateImpl::IsCertificateRevoked(const ByteSpan & certDer)
+{
+    static constexpr uint32_t maxIssuerBase64Len = BASE64_ENCODED_LEN(kMaxCertificateDistinguishedNameLength);
+
+    char issuerNameBuffer[maxIssuerBase64Len]                            = { 0 };
+    char serialNumberHexStrBuffer[2 * kMaxCertificateSerialNumberLength] = { 0 };
+    char akidHexStrBuffer[2 * kAuthorityKeyIdentifierLength]             = { 0 };
+
+    MutableCharSpan issuerName(issuerNameBuffer);
+    MutableCharSpan serialNumber(serialNumberHexStrBuffer);
+    MutableCharSpan akid(akidHexStrBuffer);
+
+    VerifyOrReturnValue(CHIP_NO_ERROR == GetIssuerNameBase64Str(certDer, issuerName), false);
+    ChipLogDetail(NotSpecified, "Issuer: %.*s", static_cast<int>(issuerName.size()), issuerName.data());
+
+    VerifyOrReturnValue(CHIP_NO_ERROR == GetSerialNumberHexStr(certDer, serialNumber), false);
+    ChipLogDetail(NotSpecified, "Serial Number: %.*s", static_cast<int>(serialNumber.size()), serialNumber.data());
+
+    VerifyOrReturnValue(CHIP_NO_ERROR == GetAKIDHexStr(certDer, akid), false);
+    ChipLogDetail(NotSpecified, "AKID: %.*s", static_cast<int>(akid.size()), akid.data());
+
+    return IsEntryInRevocationSet(akid, issuerName, serialNumber);
+}
+
+void TestDACRevocationDelegateImpl::CheckForRevokedDACChain(
+    const DeviceAttestationVerifier::AttestationInfo & info,
+    Callback::Callback<DeviceAttestationVerifier::OnAttestationInformationVerification> * onCompletion)
+{
+    AttestationVerificationResult attestationError = AttestationVerificationResult::kSuccess;
+
+    if (mDeviceAttestationRevocationSetPath != nullptr)
+    {
+        ChipLogDetail(NotSpecified, "Checking for revoked DAC in %s", mDeviceAttestationRevocationSetPath);
```

Comment:
Suggeste using std::string for the mDeviceAttestationRevocationSetPath

Reply:
Using earlier suggestion, `std::string_view`